src/store/index-level.ts

```diff
@@ -0,0 +1,205 @@
+import type { LevelDatabase } from './create-level.js';
+import type { AbstractBatchDelOperation, AbstractBatchOperation, AbstractIteratorOptions } from 'abstract-level';
+
+import { abortOr } from '../utils/abort.js';
+import { createLevelDatabase } from './create-level.js';
+import flat from 'flat';
+
+const KEY_DELIMITER = '<';
+const KEY_BOUNDARY = '>';
+
+const VALUE_DELIMITER = '=';
+
+const KEY_PREFIX_DELIMITER = '@';
+
+export type Entry = {
+  _id: string,
+  [property: string]: unknown
+};
+
+export type ExactFilter = string | number | boolean;
+
+export type RangeFilter = {
+  gte?: string,
+  lte?: string
+};
```

Comment:
@dcrousso thoughts on:

```typescript
type Gt = ({ gt: string } & { gte?: never }) |
                   ({ gt?: never } & { gte: string });
type Lt = ({ lt: string } & { lte?: never }) |
                   ({ lt?: never } & { lte: string });

export type RangeFilter = (Gt | Lt) & Partial<Gt> & Partial<Lt>;
```

* at least one of `gt`, `gte`, `lt`, or `lte` is required
* `gt` and `gte` are mutually exclusive.
* `lt` and `lte` are mutually exclusive.

has no benefit w.r.t. runtime checks because type erasure. but does provide a better developer experience